### ISSUE TITLE
Nvenc render corrupted video on arch

### ISSUE THREAD

**Issue by Zeioth** [MaartenBaert/ssr#658]
I've already posted all the details [here](https://video.stackexchange.com/questions/24100/nvenc-produces-corrupted-video).

**Comment by MaartenBaert** [comment]
Yes, I'm aware of it but I haven't been able to find the cause yet. It used to work, but some update outside of SSR seems to have broken it.

**Comment by MaartenBaert** [comment]
I suspect that there is some kind of race condition or something involving reuse of freed memory, because otherwise I can't explain why it works in the ffmpeg command-line tool but not in SSR. But these things are very hard to track down. Maybe knowing the specific version will allow me to dig through the diffs and figure out the cause - thanks!

**Comment by arrowgent** [comment]
id like to bring this back up, since its been quiet

i noticed this change specifically to ssr v4.3
previous ssr v4.2 doesnt have this corrupted video with nvenc

using the same ffmpeg version 4.3.2
ubuntu 18.04.x LTS

**Comment by MaartenBaert** [comment]
@arrowgent Are you sure that the only difference between the two SSR versions you are testing is the version number v4.2 versus v4.3, and nothing else? Because I haven't made any changes to SSR that would affect Nvenc encoding, and this issue has existed for a very long time, long before v4.2.

**Comment by arrowgent** [comment]
this was quite some time ago, december 2020 when i noticed this

recently updating ffmpeg to v4.4-1 and setting nvenc doesnt have any issues in ssr

im not sure what else
sorry for old info

**Comment by arrowgent** [comment]
simplescreenrecorder:amd64 (0.4.3+2~ppa1~bionic1, 0.4.4+1~ppa1~bionic1)
vs
ssr-master 2021-feb-06 src (4.3)
vs
ssr-master 2021-june-12 src (4.4)

so i wanted to take a moment and look at these differences


this is 4.3 compiled from src, that i have
https://github.com/MaartenBaert/ssr/commit/f7f1405cbeebf0d86d442e898d9d00795c69a4d1
![4.3](https://i.imgur.com/B5VKbla.png)

this is 4.4 updated from the PPA yesterday
![4.4](https://i.imgur.com/sBAx0O0.png)

this is 4.4 i compiled from src today.
https://github.com/MaartenBaert/ssr/commit/90ff226c0cf8dd887643aefe8f4f29f85747c100
![4.4](https://i.imgur.com/l5vogK7.png)

notice the libavformat, libavcodec, etc?
this difference must be causing NVENC capture corruption

where am i getting the newer libav?  a backport ppa that includes ffmpeg etc, doesnt matter, everything else works fine in other areas of my system (obs, mpv, video editing tools, etc)

compiling the source using system libraries allows proper NVENC capture recording
if using your PPA, with different libraries compiled than installed, this is causing a corruption in the video

**Comment by MaartenBaert** [comment]
The strange thing is that not only the headers are different, the library version is also different, which means you must have more than one version of FFmpeg installed on your system. What do you get when you run `ldd` on each of your `simplescreenrecorder` binaries? This should tell you where the libraries are being loaded from.

For example:
```
maarten@desktop-maarten ~ $ ldd /usr/bin/simplescreenrecorder  | grep libav
	libavformat.so.58 => /usr/lib/libavformat.so.58 (0x00007fba6b8da000)
	libavcodec.so.58 => /usr/lib/libavcodec.so.58 (0x00007fba6a4a1000)
	libavutil.so.56 => /usr/lib/libavutil.so.56 (0x00007fba6a1f6000)
maarten@desktop-maarten ~ $ ldd ~/Documents/ssr/build-release/src/simplescreenrecorder | grep libav
	libavformat.so.58 => /usr/lib/libavformat.so.58 (0x00007f35d6169000)
	libavcodec.so.58 => /usr/lib/libavcodec.so.58 (0x00007f35d4d30000)
	libavutil.so.56 => /usr/lib/libavutil.so.56 (0x00007f35d4a85000)
```
